azure/services/roleassignments/client.go

```diff
@@ -51,18 +60,54 @@ func newRoleAssignmentClient(subscriptionID string, baseURI string, authorizer a
 	return roleClient
 }
 
-// Create creates a role assignment.
-// Parameters:
-// scope - the scope of the role assignment to create. The scope can be any REST resource instance. For
-// example, use '/subscriptions/{subscription-id}/' for a subscription,
-// '/subscriptions/{subscription-id}/resourceGroups/{resource-group-name}' for a resource group, and
-// '/subscriptions/{subscription-id}/resourceGroups/{resource-group-name}/providers/{resource-provider}/{resource-type}/{resource-name}'
-// for a resource.
-// roleAssignmentName - the name of the role assignment to create. It can be any valid GUID.
-// parameters - parameters for the role assignment.
-func (ac *azureClient) Create(ctx context.Context, scope string, roleAssignmentName string, parameters authorization.RoleAssignmentCreateParameters) (authorization.RoleAssignment, error) {
-	ctx, _, done := tele.StartSpanWithLogger(ctx, "roleassignments.AzureClient.Create")
+// Get gets the specified role assignment by the role assignment name.
+func (ac *azureClient) Get(ctx context.Context, spec azure.ResourceSpecGetter) (interface{}, error) {
+	ctx, span := tele.Tracer().Start(ctx, "roleassignments.AzureClient.Get")
+	defer span.End()
+	raSpec := spec.(*RoleAssignmentSpec)
+	// ToDo: Check if correct scope is being passed
+	return ac.roleassignments.Get(ctx, raSpec.Scope, spec.ResourceName())
+}
+
+// CreateOrUpdateAsync creates a roleassignment.
+// Creating a roleassignment is not a long running operation, so we don't ever return a future.
+func (ac *azureClient) CreateOrUpdateAsync(ctx context.Context, spec azure.ResourceSpecGetter, parameters interface{}) (interface{}, azureautorest.FutureAPI, error) {
+	ctx, _, done := tele.StartSpanWithLogger(ctx, "groups.AzureClient.CreateOrUpdate")
 	defer done()
+	assignment, ok := parameters.(authorization.RoleAssignmentPropertiesWithScope)
+	if !ok {
+		return nil, nil, errors.Errorf("%T is not a authorization.RoleAssignment", parameters)
+	}
+	//scope := fmt.Sprintf("/subscriptions/%s/", *assignment.Scope)
+	roleAssignmentCreatePrams := authorization.RoleAssignmentCreateParameters{
```

Comment:
It's a bit awkward to need to recreate the parameters here like you said, but I think this is probably the best way to do it.

Reply:
not sure I follow, why are we not using the passed in parameters?